src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -320,37 +340,138 @@ private async Task<bool> FixTextBufferAsync(TextBufferCodeCleanUpScope textBuffe
+        {
+            progressTracker.AddItems(solution.ProjectIds.Count);
+            foreach (var projectId in solution.ProjectIds)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var project = solution.GetProject(projectId);
+                progressTracker.Description = project.Name;
+                var newProject = await FixProjectAsync(project, enabledFixIds, new ProgressTracker(), cancellationToken).ConfigureAwait(false);
+                solution = newProject.Solution;
+                progressTracker.ItemCompleted();
+            }
+
+            return solution;
+        }
+
+        private async Task<Project> FixProjectAsync(Project project, FixIdContainer enabledFixIds, ProgressTracker progressTracker, CancellationToken cancellationToken)
+        {
+            if (project.LanguageServices.GetService<ICodeCleanupService>() is null)
+            {
+                return project;
+            }
+
+            progressTracker.AddItems(project.DocumentIds.Count);
+            foreach (var documentId in project.DocumentIds)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var document = project.GetDocument(documentId);
+                progressTracker.Description = document.Name;
+                var fixedDocument = await FixDocumentAsync(document, enabledFixIds, new ProgressTracker(), cancellationToken).ConfigureAwait(false);
+                project = fixedDocument.Project;
+                progressTracker.ItemCompleted();
+            }
+
+            return project;
+        }
+
+        private async Task<Document> FixDocumentAsync(Document document, FixIdContainer enabledFixIds, ProgressTracker progressTracker, CancellationToken cancellationToken)
+        {
+            var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
+
+            var allDiagnostics = codeCleanupService.GetAllDiagnostics();
+
+            var enabedDiagnosticSets = ArrayBuilder<DiagnosticSet>.GetInstance();
+
+            foreach (var diagnostic in allDiagnostics.Diagnostics)
+            {
+                foreach (var diagnosticId in diagnostic.DiagnosticIds)
+                {
+                    if (enabledFixIds.IsFixIdEnabled(diagnosticId))
+                    {
+                        enabedDiagnosticSets.Add(diagnostic);
+                        break;
+                    }
+                }
+            }
+
+            var isRemoveUnusedUsingsEnabled = enabledFixIds.IsFixIdEnabled(RemoveUnusedImportsFixId);
+            var isSortUsingsEnabled = enabledFixIds.IsFixIdEnabled(SortImportsFixId);
+            var enabledDiagnostics = new EnabledDiagnosticOptions(enabedDiagnosticSets.ToImmutableArray(),
+                new OrganizeUsingsSet(isRemoveUnusedUsingsEnabled, isSortUsingsEnabled));
```

Comment:
same for this?

Reply:
➡️ same as https://github.com/dotnet/roslyn/pull/32175#discussion_r246928624